awx/api/views/__init__.py

```diff
@@ -4405,3 +4417,66 @@ def get_queryset(self):
         name = camelcase_to_underscore(attr)
         view = value.as_view()
         setattr(this_module, name, view)
+
+
+class WorkflowApprovalTemplateDetail(RelatedJobsPreventDeleteMixin, RetrieveUpdateDestroyAPIView):
+
+    model = models.WorkflowApprovalTemplate
+    serializer_class = serializers.WorkflowApprovalTemplateSerializer
+
+
+class WorkflowApprovalTemplateJobsList(SubListAPIView):
+
+    model = models.WorkflowApproval
+    serializer_class = serializers.WorkflowApprovalListSerializer
+    parent_model = models.WorkflowApprovalTemplate
+    relationship = 'approvals'
+    parent_key = 'workflow_approval_template'
+
+
+class WorkflowApprovalList(ListCreateAPIView):
+
+    model = models.WorkflowApproval
+    serializer_class = serializers.WorkflowApprovalListSerializer
+
+    def get(self, request, *args, **kwargs):
+        if not request.user.is_superuser and not request.user.is_system_auditor:
+            raise PermissionDenied(_("Superuser privileges needed."))
+        return super(WorkflowApprovalList, self).get(request, *args, **kwargs)
+
+
+class WorkflowApprovalDetail(UnifiedJobDeletionMixin, RetrieveDestroyAPIView):
+
+    model = models.WorkflowApproval
+    serializer_class = serializers.WorkflowApprovalSerializer
+
+
+class WorkflowApprovalApprove(RetrieveAPIView):
+    model = models.WorkflowApproval
+    serializer_class = serializers.WorkflowApprovalViewSerializer
+
+    # &&&&&& To address later
+    def post(self, request, *args, **kwargs):
+        obj = self.get_object()
```

Comment:
You probably need some code here to check the permissions:

```
can_approve = request.user.can_access(models.User, 'approve_or_deny', obj)
if not can_approve:
    raise PermissionDenied(_('Cannot approve.'))
```

Reply:
yeah, I think that's part of the TODO, I anticipate that it will look like

`request.user.can_access('approve_or_deny', obj)`

The point is that it calls the logic written in `access.py`. One way or the other, it just needs to be imported and called here. You don't _have_ to put the logic in the access file (you could put it here instead), it's just easier to follow that way